### PR TITLE
Null check before new TagService

### PR DESCRIPTION
**Pull request by Luncher91**
By ordering public shares, UserFolder and UserSession are null and lead to an Error while creating a TagService.

## Description
The UserSession and the UserFolder will be checked for being null before creating a new TagService object.

## Related Issue
None

## Motivation and Context
1. Being not logged in
2. open a public share
3. sort the files by name
-> error while loading page
4. the page reloads automatically

## How Has This Been Tested?
changed in my version from the apt repository and reproduced these steps.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: apps/files/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -81,11 +81,16 @@ public function __construct(array $urlParams= []) {
...
+			$userSession = $c->query('ServerContainer')->getUserSession();
+			
+			if(!is_null($userSession) && !is_null($homeFolder))
+			{
+				return new TagService(
+					$userSession,
+					$c->query('Tagger'),
+					$homeFolder
+				);
+			}
```
Comment (PVince81) [inline apps/files/lib/AppInfo/Application.php]: `return null` otherwise ?

**Comment by PVince81** [comment]
Tested with your steps but this PR doesn't fix the issue.

What I observe is not related to tags but the file list seems to call an API: http://localhost/owncloud/index.php/apps/files/api/v1/sorting which returns 401 Not Authenticated.

From what I see this endpoint only exists to be able to store the sorting settings when logged in.
The correct fix is to find a way to prevent sort persistence when dealing with a public page.

Need to look at https://github.com/owncloud/core/blob/v9.1.4/apps/files_sharing/js/public.js#L158 and find a way to set the "persist" always to false. Maybe overriding `setSort` could be the way to go.

**Comment by DeepDiver1975** [comment]
@Luncher91 any update on this? THX

**Comment by Luncher91** [comment]
Not yet. But I will look at the suggested files. @PVince81 considerations seem to be plausable on the first glance.

**Comment by Luncher91** [comment]
Please, can someone fix that unittest who is maybe more familiar with the intendet function?

File: apps/files/lib/AppInfo/Application.php (hunk 1 above)
Comment (PVince81) [inline apps/files/lib/AppInfo/Application.php]: ping

File: apps/files/lib/AppInfo/Application.php [hunk 2]
```diff
@@ -81,11 +81,16 @@ public function __construct(array $urlParams= []) {
...
 			$homeFolder = $c->query('ServerContainer')->getUserFolder();
-			return new TagService(
-				$c->query('ServerContainer')->getUserSession(),
-				$c->query('Tagger'),
-				$homeFolder
-			);
+			$userSession = $c->query('ServerContainer')->getUserSession();
+			
+			if(!is_null($userSession) && !is_null($homeFolder))
+			{
```
Comment (PVince81) [inline apps/files/lib/AppInfo/Application.php]: move this on the previous line, see https://doc.owncloud.org/server/10.0/developer_manual/general/codingguidelines.html#coding about braces

File: apps/files/js/filelist.js [hunk 3]
```diff
@@ -1534,7 +1534,7 @@
 				}
 			}
 
-			if (persist) {
+			if (persist && OC.getCurrentUser().uid != null) {
```
Comment (PVince81) [inline apps/files/js/filelist.js]: change to `!==` for strict equality, see https://doc.owncloud.org/server/10.0/developer_manual/general/codingguidelines.html#id1

**Comment by PVince81** [comment]
No one can push on your fork, you'll have to fix it yourself.

Try running: `make test-js-debug`.
Then patch fileListSpec.js:
```patch
diff --git a/apps/files/tests/js/filelistSpec.js b/apps/files/tests/js/filelistSpec.js
index c5186a65c7..b9ef3f39d7 100644
--- a/apps/files/tests/js/filelistSpec.js
+++ b/apps/files/tests/js/filelistSpec.js
@@ -2299,6 +2299,14 @@ describe('OCA.Files.FileList tests', function() {
                });
        });
        describe('Sorting files', function() {
+               var currentUserStub;
+
+               beforeEach(function() {
+                       currentUserStub = sinon.stub(OC, 'getCurrentUser').returns({uid: 'user1'});
+               });
+               afterEach(function() {
+                       currentUserStub.restore();
+               });
                it('Toggles the sort indicator when clicking on a column header', function() {
                        var ASC_CLASS = fileList.SORT_INDICATOR_ASC_CLASS;
                        var DESC_CLASS = fileList.SORT_INDICATOR_DESC_CLASS;
```

**Comment by PVince81** [comment]
@Luncher91 any update ?

**Comment by Luncher91** [comment]
is there anything wrong with the build system (mysql docker timed out?) or did my fork actually fail?

**Comment by PVince81** [comment]
it's likely that the build system is moody today

**Comment by PVince81** [comment]
:+1: tested, works.

**Pull request merged**